addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/internal/SenseBoxAPIConnection.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.sensebox.internal;
+
+import com.google.gson.Gson;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.sensebox.model.SenseBoxData;
+import org.openhab.binding.sensebox.model.SenseBoxDescriptor;
+import org.openhab.binding.sensebox.model.SenseBoxLoc;
+import org.openhab.binding.sensebox.model.SenseBoxLocation;
+import org.openhab.binding.sensebox.model.SenseBoxSensor;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.Version;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+
+import static org.openhab.binding.sensebox.SenseBoxBindingConstants.INVALID_BRIGHTNESS;
+import static org.openhab.binding.sensebox.SenseBoxBindingConstants.SENSEMAP_API_URL_BASE;
+import static org.openhab.binding.sensebox.SenseBoxBindingConstants.SENSEMAP_IMAGE_URL_BASE;
+import static org.openhab.binding.sensebox.SenseBoxBindingConstants.SENSEMAP_MAP_URL_BASE;
+
+/**
+ * The {@link SenseBoxAPIConnection} is responsible for fetching data from the senseBox API server.
+ *
+ * @author Hakan Tandogan - Initial contribution
+ */
+public class SenseBoxAPIConnection {
+    private Logger logger = LoggerFactory.getLogger(SenseBoxAPIConnection.class);
+
+    private Gson gson = new Gson();
+
+    private final static Properties HEADERS = new Properties();
+
+    private final static String METHOD = "GET";
+
+    private final static int TIMEOUT = 30 * 1000; // 30 seconds
+
+    public SenseBoxAPIConnection() {
+        String openHABVersion = "2.1.0-SNAPSHOT";
+        Version version = FrameworkUtil.getBundle(this.getClass()).getVersion();
+        // TODO is there a way to obtain the OH version?
```

Comment:
Yes, there is, I did not find a way to obtain the OH version programmatically. I'm typing a question about this in the forums just right now 😄